packages/explorer/src/common.ts

```diff
@@ -1,22 +1,36 @@
 import { Chain, anvil, garnet, redstone } from "viem/chains";
 
-export const chains: Partial<Record<number, Chain>> = {
+const supportedChains = [anvil, garnet, redstone];
+
+export type SupportedChainIds = (typeof supportedChains)[number]["id"];
+export type SupportedChainNames = "anvil" | "garnet" | "redstone";
+
+export const chainIdName: Record<SupportedChainIds, SupportedChainNames> = {
+  [anvil.id]: "anvil",
+  [garnet.id]: "garnet",
+  [redstone.id]: "redstone",
+} as const;
+
+export const chainNameId: Record<SupportedChainNames, SupportedChainIds> = {
+  anvil: anvil.id,
+  garnet: garnet.id,
+  redstone: redstone.id,
+};
+
+export const chains: Record<SupportedChainIds, Chain> = {
   [anvil.id]: anvil,
-  [redstone.id]: redstone,
   [garnet.id]: garnet,
+  [redstone.id]: redstone,
 };
```

Comment:
alternatively could name the chains here like

```ts
const supportedChains = { redstone, garnet, anvil };
```

then compute the look ups based on that, or do look ups like

```ts
const chain = Object.values(supportedChains).find((c) => c.id === chainId);
```